test/unit/collaborators.test.js

```diff
@@ -1,16 +1,120 @@
+      });
+
+      it('should return false if the collaborator type is not TSC', () => {
+        // Collaborator 'foo' is not TSC
+        const collaboratorNoTSC = collaborators.get('foo');
+        assert.strictEqual(collaboratorNoTSC.isTSC(), false);
+      });
+    });
+
+    describe('getName', () => {
+      it('should return the name and login', () => {
+        collaborators.forEach(collaborator => {
+          assert.strictEqual(collaborator.getName(), `${collaborator.name}(${collaborator.login})`);
+        });
+      });
+    });
+
+    describe('getContact', () => {
+      it('should return the name and the email', () => {
+        collaborators.forEach(collaborator => {
+          assert.strictEqual(collaborator.getContact(), `${collaborator.name} <${collaborator.email}>`);
+        });
+      });
+    });
+  });
+
+  describe('getCollaborators', () => {
+    it('should return all collaborators', async function() {
+      const parsed = await getCollaborators(readme, logger, 'nodejs', 'node');
+      assert.deepStrictEqual(parsed, collaborators);
+    });
+
+    it('should throw error if there is no TSC section in the README', async() => {
+      await assertThrowsAsync(async() => getCollaborators(readmeNoTsc, logger, 'nodejs', 'node'), /Error: Couldn't find ### TSC \(Technical Steering Committee\) in the README/);
+    });
+
+    it('should throw error if there is no TSC Emeriti section in the README', async() => {
+      await assertThrowsAsync(async() => getCollaborators(readmeNoTscE, logger, 'nodejs', 'node'), /Error: Couldn't find ### TSC Emeriti in the README/);
+    });
+
+    it('should throw error if there is no Collaborators section in the README', async() => {
+      await assertThrowsAsync(async() => getCollaborators(readmeNoCollaborators, logger, 'nodejs', 'node'), /Error: Couldn't find ### Collaborators in the README/);
+    });
+
+    it('should throw error if there is no Collaborator Emeriti section in the README', async() => {
+      await assertThrowsAsync(async() => getCollaborators(readmeNoCollaboratorE, logger, 'nodejs', 'node'), /Error: Couldn't find ### Collaborator Emeriti in the README/);
+    });
+
+    it('should WARN if the TSC and Collaborators section are not ordered in the README', async() => {
+      await assertThrowsAsync(async() => getCollaborators(readmeUnordered, logger, 'nodejs', 'node'), /Error/);
+      assert.strictEqual(logger.logs.warn[0][0], 'Order of different types of contacts needs update. Analysis could get wrong.');
+    });
+
+    it('should throw error if there are no collaborators', async() => {
+      await assertThrowsAsync(async() => getCollaborators(readmeUnordered, logger, 'nodejs', 'node'), /Error: Could not find any collaborators/);
+    });
+  });
+
+  describe('isCollaborator', () => {
+    it('should return undefined if the user has no login (ghost)', () => {
+      assert.strictEqual(isCollaborator(collaborators, { name: 'Ghost' }), undefined);
```

Comment:
If it's a ghost then the whole thing would be `null`, the login check is just a safeguard, I don't think Github would ever return an actor without `login`..